src/js/svgo-worker/index.js

```diff
@@ -1,270 +1,98 @@
+                const viewBox = node.attributes.viewBox.split(/(?:,\s*|\s+)/);
+                dimensions.width = Number.parseFloat(viewBox[2]);
+                dimensions.height = Number.parseFloat(viewBox[3]);
+              }
+            }
+          }
+        }
+      };
     }
-    else if (
-      obj.constructor === CSSClassList ||
-      obj.constructor === CSSStyleDeclaration ||
-      obj.constructor === Object ||
-      obj.constructor === Array
-    ) {
-      objClone = new obj.constructor();
-      clones.set(obj, objClone);
-      cloneKeys(objClone, obj);
-    }
-    else if (obj.constructor === Map) {
-      objClone = new Map();
-      clones.set(obj, objClone);
-
-      for (const [key, val] of obj) {
-        objClone.set(clone(key), clone(val));
-      }
-    }
-    else if (obj.constructor === Set) {
-      objClone = new Set();
-      clones.set(obj, objClone);
-
-      for (const val of obj) {
-        objClone.add(clone(val));
-      }
-    }
-    else {
-      throw Error('unexpected type');
-    }
-
-    return objClone;
-  }
-
-  return clone(svg);
-}
-
-// Arrange plugins by type - this is what plugins() expects
-function optimizePluginsArray(plugins) {
-  return plugins.map(item => [item]).reduce((arr, item) => {
-    const last = arr[arr.length - 1];
-
-    if (last && item[0].type === last[0].type) {
-      last.push(item[0]);
-    }
-    else {
-      arr.push(item);
-    }
-    return arr;
-  }, []);
-}
-
-const optimisedPluginsData = optimizePluginsArray(Object.values(pluginsData));
-
-function getDimensions(parsedSvg) {
-  const svgEl = parsedSvg.content.filter(el => el.isElem('svg'))[0];
-
-  if (!svgEl) {
-    return {};
-  }
-
-  if (svgEl.hasAttr('width') && svgEl.hasAttr('height')) {
-    return {
-      width: parseFloat(svgEl.attr('width').value),
-      height: parseFloat(svgEl.attr('height').value)
-    };
-  }
-
-  if (svgEl.hasAttr('viewBox')) {
-    const viewBox = svgEl.attr('viewBox').value.split(/(?:,\s*|\s+)/);
-
-    return {
-      width: parseFloat(viewBox[2]),
-      height: parseFloat(viewBox[3])
-    };
-  }
-
-  return {};
-}
+  };
+  return [dimensions, plugin];
+};
 
 function* multipassCompress(settings) {
-  // activate/deactivate plugins
-  Object.keys(settings.plugins).forEach(pluginName => {
-    pluginsData[pluginName].active = settings.plugins[pluginName];
-  });
-
-  // Set floatPrecision across all the plugins
+  // setup plugin list
   const floatPrecision = Number(settings.floatPrecision);
-
-  for (const plugin of Object.values(pluginsData)) {
-    if (plugin.params && 'floatPrecision' in plugin.params) {
-      if (plugin === pluginsData.cleanupNumericValues && floatPrecision === 0) {
+  const plugins = [];
+  for (const [pluginName, active] of Object.entries(settings.plugins)) {
+    if (active) {
+      const plugin = {
+        name: pluginName,
+        params: {}
+      };
+      if (plugin.name === 'cleanupNumericValues' && floatPrecision === 0) {
```

Comment:
I'm interested to hear your thoughts on this. Is this something SVGO should be doing?

Reply:
This plugin is too generic. I'd prefer to have cleanup as part of other plugins. Though it is still with us for now.